Plugins (wpcom): Update design and text

drw158 opened the pull request:
This addresses some feedback. fixes #4986 

1. Make the plugins more active by making them green.
2. Update text.
3. Make it more obvious that the links are external.

@dmsnell For 3, can we change the plugin icon on hover, if the link is external? Currently this is all but the Google Analytics plugin. My idea is to change the plugin icon to the `external` one on hover.

Another solution would be to add a tooltip for each link.

### Before:

![obhyaltluaaaaasuvork5cyii](https://cloud.githubusercontent.com/assets/618551/14821456/6ea2131c-0b90-11e6-8df1-bc95882b902e.png)

### After:

![acyhettljyusqaaaabjru5erkjggg](https://cloud.githubusercontent.com/assets/618551/14797800/15e3961a-0afa-11e6-8dd0-45689ba8c79a.png)

Hover state:

![screen shot 2016-04-26 at 9 21 34 am](https://cloud.githubusercontent.com/assets/618551/14821422/5332cf7c-0b90-11e6-98a2-250f8aaddbea.png)

drw158 commented:
Thanks Dennis, the icon replacement is working well.

dmsnell commented:
@drw158 I have merged the plugin components and now this PR has more red than green!

folletto commented:
All looks good to me design wise. 

The only doubt I have is this:

folletto commented:
All looks good to me design wise. 

The only doubt I have is this:

> Make the plugins more active by making them green.

Can I ask why green instead of our _medium blue_ which seems equally "highlighting" but at the same time more aligned with the overall page look?

drw158 commented:
First iteration was blue, but then it looked like "too much blue." Second iteration was gray as shown in the "before" screenshot.

We received feedback that the active plugins may not look very active, so I changed it to green. I also thought it was a good opportunity to introduce some color other than blue.

But, if the green does not look good, I'd be fine with going back to gray.

mtias commented:
@drw158 wonder if this is a chance to try the color-code @adambbecker was exploring for plans (a color for each plan, basically).

drw158 commented:
Worth a shot! @adambbecker can I just pick the colors from the screenshot on your mega Plans post a few weeks ago?

folletto commented:
> First iteration was blue, but then it looked like "too much blue." Second iteration was gray as shown in the "before" screenshot.

Between "too much blue" and "too much green" I'd go for blue given the "not active enough" feedback, which is fair. 
Just my take, it's ok if we decide to rather go green. :)

dmsnell commented:
> I'd go for blue given the "not active enough" feedback

yeah the whole idea of adding color here because it's "too blue" always seemed strange to me. blue is our entire app's color scheme…

gwwar commented on client/my-sites/plugins-wpcom/plugin.jsx:
```diff
@@ -18,11 +28,29 @@ export const StandardPlugin = React.createClass( {
 
+		const { isUnderMouse } = this.state;
+
+		const isExternalLink = hasHttpProtocol( descriptionLink );
```
Can we add this as a prop, instead of testing for this?

dmsnell replied:
@gwwar we deliberated on this point and I think it's best to keep the logic contained inside this component. it's only effect is visual and I don't think there's any need to couple that to the "outside world" I like being able to only care about adding the URL in the plugin data structure instead of having to think about how to represent that data.

gwwar commented on client/my-sites/plugins-wpcom/premium-plugins-panel.jsx:
```diff
@@ -26,9 +30,10 @@ export const PremiumPluginsPanel = React.createClass( {
-								{ ...{ name, key: name, descriptionLink, icon, plan, description } }
+						{ plugins.map( ( { name, descriptionLink, icon, category, description } ) =>
+							<Plugin
+								onClick={ () => onClick( name ) }
```
We can avoid a fat arrow here, if we move the default analytics tracking down a level to the Plugin component. The Plugin component will have name available as a prop.

dmsnell replied:
This was difficult. The plugin component is the same regardless of the plan level, so the analytics code needs to know whether it's for a standard, premium, or business plugin. I tried hard to eliminate the fat-arrow function but didn't find any good way to do so. Granted, I _could_ eliminate it _per se_, but the underlying process (creating a new closure) was going to happen regardless. I chose to keep it here where it was obvious to someone reading the code.

For example, I could have made `onClick = name => () => dispatch( tracking( name ) )` but that would hide what's happening and I think that takes a step away from what we want semantically to happen.

Further, I didn't want to arbitrarily add props to the plugin piece itself or its `onClick` function because I felt that the _meaning_ of `onClick` should be just that: a generic function that gets called when the plugin is clicked, verses some specialized function taking specific parameters.

Feel free to disagree and point out why you think this is wrong.

gwwar replied:
:) The fat arrow warnings are non-blocking, but its just nice if we can avoid them.

Another alternative is to let the plugin provide an interface, where the plugin prop values are provided, and we pass through a trackClick function.

In Plugin:
```
onClick() {
     this.props.trackClick( this.props );
     this.props.onClick();
}
```

gwwar commented on client/my-sites/plugins-wpcom/default-plugins.js:
```diff
@@ -165,9 +165,9 @@ export const defaultPremiumPlugins = [
 	{
 		name: i18n.translate( 'Google Analytics' ),
-		descriptionLink: '/plans/features/google-analytics/{siteSlug}',
+		descriptionLink: '/categorys/features/google-analytics/{siteSlug}',
```
categories?

dmsnell replied:
question marks?

not sure I understand what you are commenting here @gwwar :wink:

maybe this is a typo and it needs to remain `plans` - I did a find/replace so that's probably it

gwwar commented on client/my-sites/plugins-wpcom/standard-plugins-panel.jsx:
```diff
@@ -43,4 +48,17 @@ StandardPluginsPanel.propTypes = {
+);
+
+const mapDispatchToProps = dispatch => ( {
+	onClick: name => dispatch( trackClick( name ) )
```
```javascript
const mapDispatchToProps = ( dispatch, props ) => ( {
    onClick: () => dispatch( trackClick( props.name ) )
} ); 
```

gwwar commented on client/my-sites/plugins-wpcom/standard-plugins-panel.jsx:
```diff
@@ -27,7 +31,8 @@ export const StandardPluginsPanel = React.createClass( {
 						{ shownPlugins.map( ( { name, descriptionLink, icon, category, description } ) =>
-							<StandardPlugin
+							<Plugin
+								onClick={ () => onClick( name ) }
```
If [we add this](https://github.com/Automattic/wp-calypso/pull/4995/files/70ba8c8eca897730bbdc82700b0771359c226790#r61158480), then we can avoid the fat arrow here:
```javascript
onClick={ onClick }
```

gwwar commented:
👍 Codewise this is good to ship, after we change back the analytics page slug to `/plans/features/google-analytics/{siteSlug}`

Are we okay with having a blue hover state? This kind of stuck out to me:
<img width="238" alt="screen shot 2016-04-26 at 2 03 04 pm" src="https://cloud.githubusercontent.com/assets/1270189/14834193/f65c8356-0bb7-11e6-9a89-f0ef8ce7fb65.png">

drw158 commented:
Blue hover state was intentional, just to emulate a blue text link. I didn't think about doing a lighter green though. I'm not sure where I stand on the color in general. Not sure whether to do green, blue, or gray.

adambbecker commented:
@mtias, @drw158: Ah, yeah I see what you mean about the plan colors. I think it could work, but I agree with Dave that the yellow might look a little like a warning. I'm also still playing around with trying those colors outside of the plans page & unfortunately the connection just isn't very strong between remembering those colors & associating them with plans (until you see all the plans on the page).

Seeing this action, I actually really dig the green because it enforces the active state, but I can also see what others have mentioned that it can be a little much once a lot of plugins are visible. I was playing around a bit here just to see what some other versions could look like and I settled on something like this:

![screen shot 2016-04-26 at 5 53 55 pm](https://cloud.githubusercontent.com/assets/1427136/14835455/e11a854a-0bd7-11e6-8bb6-c1c1416df200.png)

Take that with a grain of salt mostly because I was just testing things out. But a few things that caught my eye were:

* **Active state**: wanted to try the green idea, but visually scaled down a bit.
* **"Purchase" button**: when I first saw it, I was a little afraid that clicking it would automatically send me down a checkout path, so I wanted to see what an informative nudge version would look like.

---

This is really great though. All of the above is totally up to your discretion, just throwing a few more ideas into the pot just in case you were still exploring.

Overall, I do like the green direction. I would also probably remove the blue bar underneath the section header and keep the hover color the same in that particular case. Or, keep the color bar beneath the section header, but make it dynamic so that active sections get green, purchasable sections get blue.

folletto commented:
The green check marks I feel work really really well. We might get away also with the `✓ Active` label in the bar with that. Feels clear enough already. :)

mtias commented:
> I would also probably remove the blue bar underneath the section header and keep the hover color the same in that particular case.

👍  to this. The notion of "primary SectionHeader" also didn't make much sense to me.

drw158 commented:
The blue bar underneath was to address feedback that the Premium and Business sections did not stand out as "headers", and visually got lost. Currently we don't have a standard way to do something like this. But, if you feel that the header stand out enough, then I'm fine with removing it.

drw158 commented:
I also like the green check marks. When I first saw the nudge component in the mockup, I thought that was a really interesting idea. It sort of makes sense in the Calypso context. After letting it sink it though, it does look strange in the context of what a "normal" website looks like.

drw158 commented:
Ok, I changed the active state to blue for now. I will explore the check marks in another PR.

**I updated from master, so no more conflicts. Are we ok to merge now?**

roundhill commented:
> I updated from master, so no more conflicts. Are we ok to merge now?

:shipit:
